libraries/commerce/product/streams/index.js

```diff
@@ -122,3 +124,22 @@ export const productRelationsReceived$ =
 
 export const productNotAvailable$ =
   main$.filter(({ action }) => action.type === PRODUCT_NOT_AVAILABLE);
+
+/**
+ * Buffer PROVIDE_PRODUCT action and map to have array of requested products
+ * @type {Observable}
+ */
+export const fetchProductsRequested$ = main$
+  .filter(({ action }) => action.type === PROVIDE_PRODUCT)
+  .bufferTime(PROVIDE_PRODUCT_BUFFER_TIME)
+  .filter(actions => actions.length > 0)
+  .map((actions) => {
+    const productIds = actions.map(({ action: { productId } }) => productId);
+    return {
+      ...actions[0],
+      action: {
+        ...actions[0].action,
+        productId: productIds,
```

Comment:
`productId` is correct. it can be 1 product and multiple products. libraries/tracking/subscriptions/favorites.js Tracking adjusted to send multiple items to GA as well
```
case ADD_PRODUCT_TO_FAVORITES: {
      const adds = [].concat(action.productId);
```